### ISSUE TITLE
Version 0.19.0 broke some command reuse scenarios

### ISSUE THREAD

**Issue by MaceWindu** [DataAction/AdoNetCore.AseClient#194]
**Describe the bug**
Reuse of command object with same parameter name but different type doesn't update parameter formatter which leads to wrong value sent to server. This `if` is skipped: https://github.com/DataAction/AdoNetCore.AseClient/pull/184/files#diff-fa5f14f3952f93eefdf1f7f2439b1232R673-R676

**To Reproduce**
```cs
using (var cmd = cn.CreateCommand())
                {
                    var sql = @"SELECT TOP 1 Convert(Decimal(29,10), @value)";
                    var p = cmd.CreateParameter();
                    p.ParameterName = "@value";
                    p.DbType = System.Data.DbType.Decimal;
                    p.Value = 6579.64648m;
                    cmd.CommandText = sql;
                    cmd.Parameters.Add(p);

                    // 6579.64648m sent to server
                    using (var rd = cmd.ExecuteReader(System.Data.CommandBehavior.Default))
                    {
                        rd.Read();
                        var result = rd.GetDecimal(0);
                    }

                    p = cmd.CreateParameter();
                    p.ParameterName = "@value";
                    p.DbType = System.Data.DbType.Single;
                    p.Value = 6579.64648f;
                    cmd.Parameters.Clear();
                    cmd.Parameters.Add(p);

                    // decimal formatter applied to 6579.64648f and returned value is 65.79....
                    using (var rd = cmd.ExecuteReader(System.Data.CommandBehavior.Default))
                    {
                        rd.Read();
                        var result = rd.GetDecimal(0);
                    }
                }
```

**Expected behavior**
Formatter creation logic should properly test parameter

**Environment**
- .NET Framework/Core version: netfx and netcore (2.1 and 3.1)
- `AdoNetCore.AseClient` nuget package version (e.g. `0.19.0`)

**Comment by ngvtien** [comment]
Thanks for the feedback, I will fix this asap.

**Comment by ngvtien** [comment]
Addressed this issue in 0.19.1
